doc/api/typescript.md

```diff
@@ -85,6 +85,7 @@ but we recommend version 5.7 or newer with the following `tsconfig.json` setting
      "module": "nodenext",
      "allowImportingTsExtensions": true,
```

Comment:
I think rewrite ts import implies allow importing

Reply:
I think you are correct based on [this repro.](https://www.typescriptlang.org/dev/bug-workbench/?target=99&lib=lib.esnext.d.ts&rewriteRelativeImportExtensions=true&ts=5.9.0-dev.20250302#code/PTAEAEBcEMCcHMCmkBcpEGcB2iAekAoECAGwEsAjNTHfIscAE0QGMS5pIyB7LNSWAFdE9CLEQB3WGUiIASonZcAbogCSAWwAO3WJACi+RFgw8T-ISLLbdkUAG9QAM27dQAX2exuG0ACIAOmBoAMgMPwIXbgBuAlFwJzISY2gNRDQQsII8HT1QaAwATywWZ0ESrl5nVwAKAEoHAlBQcUhBWCxQABYAJlj3AiA)

Shall I remove that in this same PR?

Reply:
Yeah since we are updating the json file

Reply:
I don't remember having this implication; it's not documented on the website.

Reply:
(but there's no errors in that link so maybe it works... I'll check the code in a bit)

Reply:
Yeah, there is an implication, it's just not documented:

```ts
    allowImportingTsExtensions: {
        dependencies: ["rewriteRelativeImportExtensions"],
        computeValue: compilerOptions => {
            return !!(compilerOptions.allowImportingTsExtensions || compilerOptions.rewriteRelativeImportExtensions);
        },
    },
```

Reply:
I sent https://github.com/microsoft/TypeScript-Website/pull/3347.